tests/functional/test_install_direct_url.py

```diff
@@ -21,19 +22,21 @@ def test_install_vcs_editable_no_direct_url(script):
 
 
 @pytest.mark.usefixtures("with_wheel")
-def test_install_vcs_non_editable_direct_url(script):
+def test_install_vcs_non_editable_direct_url(script: PipTestEnvironment) -> None:
     pkg_path = _create_test_package(script, name="testpkg")
     url = path_to_url(pkg_path)
     args = ["install", f"git+{url}#egg=testpkg"]
     result = script.pip(*args)
     direct_url = get_created_direct_url(result, "testpkg")
     assert direct_url
     assert direct_url.url == url
+    assert isinstance(direct_url.info, VcsInfo)
+    assert direct_url.info
```

Comment:
I think only one of these is needed?

Reply:
Fixed in https://github.com/pypa/pip/pull/10464/commits/10d3c223de94a6c99614c662b90f4eb2471d4790.